### PR TITLE
fix: chunk memory leak, bug fix

### PR DESCRIPTION
**Pull request by mihir20**
raising a fix as mentioned in: https://github.com/samber/lo/issues/438

### CODE REVIEW

File: slice.go [hunk 1]
```diff
@@ -182,7 +182,7 @@ func Chunk[T any, Slice ~[]T](collection Slice, size int) []Slice {
 		if last > len(collection) {
 			last = len(collection)
 		}
-		result = append(result, collection[i*size:last])
+		result = append(result, collection[i*size:last:last])
```
Comment (ccoVeille) [inline slice.go:185]: I had never faced this notation, with the capacity apparently

File: slice_test.go [hunk 2]
```diff
@@ -235,6 +235,12 @@ func TestChunk(t *testing.T) {
 	allStrings := myStrings{"", "foo", "bar"}
 	nonempty := Chunk(allStrings, 2)
 	is.IsType(nonempty[0], allStrings, "type preserved")
+
+	// appending to a chunk should not affect original array
+	originalArray := []int{0, 1, 2, 3, 4, 5}
+	result5 := Chunk(originalArray, 2)
+	result5[0] = append(result5[0], 6)
+	is.Equal(originalArray, []int{0, 1, 2, 3, 4, 5})
```
Comment (ccoVeille) [inline slice_test.go:243]: So. it's not only a fix of a possible memory leak, but also a functional fix, no?

I'm unsure why you are adding this check.no computer with me to check.

My point is that the commit message could mention a memory leak fix + bug fix, maybe

**Review by ccoVeille — APPROVED** [review]
LGTM 👍

File: slice_test.go (hunk 2 above)
Comment (mihir20) [reply slice_test.go:243]: yeah, updated the title

File: slice.go (hunk 1 above)
Comment (samber) [inline slice.go:185]: Why not `collection[i*size:last:size]` ?

File: slice_test.go (hunk 2 above)
Comment (ccoVeille) [reply slice_test.go:243]: Thanks

**Comment by samber** [comment]
I would recommend using `copy()` instead of a pointer to the same memory space. I think this is the spirit of this library. WDYT?

It consumes more memory, but it seems safer to me.

**Comment by mihir20** [comment]
> I would recommend using `copy()` instead of a pointer to the same memory space. I think this is the spirit of this library. WDYT?
> 
> It consumes more memory, but it seems safer to me.

using copy will be slower compared to current implementation:
Benchmark using copy:
```
BenchmarkChunk
BenchmarkChunk/strings_10
BenchmarkChunk/strings_10-12         	14886344	        80.03 ns/op
BenchmarkChunk/strings_100
BenchmarkChunk/strings_100-12        	 1814036	       660.8 ns/op
BenchmarkChunk/strings_1000
BenchmarkChunk/strings_1000-12       	  183382	      6143 ns/op
BenchmarkChunk/ints10
BenchmarkChunk/ints10-12             	20346310	        57.41 ns/op
BenchmarkChunk/ints100
BenchmarkChunk/ints100-12            	 2672658	       433.6 ns/op
BenchmarkChunk/ints1000
BenchmarkChunk/ints1000-12           	  312517	      3871 ns/op
```
Benchmark without using copy:
```
BenchmarkChunk
BenchmarkChunk/strings_10
BenchmarkChunk/strings_10-12         	52708929	        22.42 ns/op
BenchmarkChunk/strings_100
BenchmarkChunk/strings_100-12        	12681414	        93.45 ns/op
BenchmarkChunk/strings_1000
BenchmarkChunk/strings_1000-12       	 2022960	       599.6 ns/op
BenchmarkChunk/ints10
BenchmarkChunk/ints10-12             	52086441	        22.50 ns/op
BenchmarkChunk/ints100
BenchmarkChunk/ints100-12            	12925526	        90.06 ns/op
BenchmarkChunk/ints1000
BenchmarkChunk/ints1000-12           	 1978053	       591.1 ns/op
```

File: slice.go (hunk 1 above)
Comment (mihir20) [reply slice.go:185]: ref: https://stackoverflow.com/questions/12768744/re-slicing-slices-in-golang/18911267#18911267

**Comment by samber** [comment]
I will discuss it in another issue.

**Pull request merged**